src/FixedDataTableCellGroup.js

```diff
@@ -138,38 +104,117 @@ class FixedDataTableCellGroupImpl extends React.Component {
+        {nonVirtualizedCells}
       </div>
     );
   }
 
-  _renderCell = (
-    /*number*/ rowIndex,
-    /*number*/ height,
-    /*object*/ columnProps,
-    /*object*/ cellTemplate,
-    /*number*/ left,
-    /*boolean*/ visible,
-    /*string*/ key,
-    /*number*/ columnGroupWidth,
-    /*boolean*/ isColumnReordering
-  ) /*object*/ => {
+  _computeVirtualizedCells = ({
+    /*boolean*/isColumnReordering,
+  }) => {
+    const { columns, columnsToRender, isScrolling } = this.props;
+
+    if (isScrolling) {
+      // We are scrolling, so there's no need to display any cells which lie outside the viewport.
+      // We still need to render them though, so as to not cause any unmounts. So we'll set visible
+      // as false to everyone in _staticCellArray (which also hides previously rendered cells)
+      // later on, we'll update the visibility for cells inside the viewport
+      this._staticCellArray.forEach((cell, i) => {
+        this._staticCellArray[i] = React.cloneElement(this._staticCellArray[i], { visible: false });
+      });
+    } else {
+      // reset the static cell array if scrolling is stopped
+      // this is done so that only cells inside the buffer are considered for vertical scrolling
+      this._staticCellArray = [];
+    }
+
+    for (let i = 0; i < (columnsToRender || columns).length; i++) {
+      let columnIndex = columnsToRender === undefined ? i : columnsToRender[i];
+      if (columnIndex === undefined) {
+        continue;
+      }
+
+      // if cell can't be recycled, then we can't do virtualization since the cell will be
+      // rendered all the time (by computeNonVirtualizedCells)
+      if (!columns[columnIndex].props.allowCellsRecycling) {
+        continue;
+      }
+
+      this._staticCellArray[i] = this._renderCell({
+        columnIndex,
+        staticIndex: i,
+        columnGroupWidth: 0,
+        isColumnReordering,
+      });
+    }
+  };
+
+  _computeNonVirtualizedCells = ({
+    /*boolean*/isColumnReordering,
+    /*number*/columnGroupWidth,
+  }) => {
+    const { allowColumnVirtualization, columns, columnOffsets, left } = this.props;
+
+    // no need to compute non virtualized cells if column virtualization turned on
+    if (allowColumnVirtualization) {
+      return [];
+    }
+
+    const nonVirtualizedCells = [];
+
+    for (let i = 0; i < columns.length; i++) {
+      const { allowCellsRecycling, width } = columns[i].props;
+
+      // if allowCellsRecycling was true, then it would have be included by this._staticCellArray
+      if (allowCellsRecycling) {
+        continue;
+      }
+
+      const currentPosition = columnOffsets[i];
+      const visible = currentPosition - left <= width && currentPosition - left + width >= 0;
+
+      nonVirtualizedCells[i] = this._renderCell({
+        columnIndex: i,
+        staticIndex: i,
+        columnGroupWidth,
+        isColumnReordering,
+      });
+    }
+    return nonVirtualizedCells;
+  };
+
+  _renderCell = ({
```

Comment:
refactored here

Makes more sense to just pass the column index to get the cell (earlier we passed wayyyy too many props). 

Props like columnGroupWidth and isColumnReoredering are still passed, since they're a bit expensive to recompute for every cell (and also is constant), so passed it from the caller instead.

`staticIndex` determines the key of the cell used by React. This is where column virtualization really kicks. Since `staticIndex` is determined by our `columnBufferSet`, we introduce minimal key changes.